src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/GetBuildManifestMetadataTests.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Linq;
+using System.Threading;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Microsoft.DotNet.Maestro.Tasks.Tests
+{
+    [TestFixture]
+    public class GetBuildManifestMetadataTests
+    {
+        private PushMetadataToBuildAssetRegistry pushMetadata;
+        private string testManifestFolderPath = @".\TestManifests\"; 
+
+        public GetBuildManifestMetadataTests()
+        {
+            pushMetadata = new PushMetadataToBuildAssetRegistry();
+        }
+        [Test]
+        public void EmptyManifestFolderPath()
+        {
+            Action act = () => pushMetadata.GetBuildManifestsMetadata("", new CancellationToken());
+            act.Should().Throw<ArgumentException>().WithMessage("The path is empty. (Parameter 'path')");
```

Comment:
I probably wouldn't assert the message. The fact that it throws is probably good enough, and string comparing leads to very rigid tests. :-)  The odds of the string being _wrong_ and that mattering to a caller (who also shouldn't be parsing the string message) is pretty low, and the odds of us wanting a more helpful error message is higher.